// aws-amplify-angular/__ivy_ngcc__/dist/src/components/storage/photo-picker-component/photo-picker.factory.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, ViewChild, ComponentFactoryResolver, Output, EventEmitter, } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { PhotoPickerIonicComponent } from './photo-picker.component.ionic';
import { PhotoPickerComponentCore } from './photo-picker.component.core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../directives/dynamic.component.directive';

function PhotoPickerComponent_ng_template_1_Template(rf, ctx) { }
var PhotoPickerComponent = /** @class */ (function () {
    function PhotoPickerComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.picked = new EventEmitter();
        this.loaded = new EventEmitter();
        this.uploaded = new EventEmitter();
    }
    PhotoPickerComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    PhotoPickerComponent.prototype.ngOnDestroy = function () { };
    PhotoPickerComponent.prototype.loadComponent = function () {
        var _this = this;
        var photoPickerComponent = this.framework && this.framework.toLowerCase() === 'ionic'
            ? new ComponentMount(PhotoPickerIonicComponent, {
                url: this.url,
                path: this.path,
                storageOptions: this.storageOptions,
            })
            : new ComponentMount(PhotoPickerComponentCore, {
                url: this.url,
                path: this.path,
                storageOptions: this.storageOptions,
            });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(photoPickerComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = photoPickerComponent.data;
        componentRef.instance.picked.subscribe(function (e) {
            _this.picked.emit(e);
        });
        componentRef.instance.loaded.subscribe(function (e) {
            _this.loaded.emit(e);
        });
        componentRef.instance.uploaded.subscribe(function (e) {
            _this.uploaded.emit(e);
        });
    };
    /** @nocollapse */
    PhotoPickerComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    PhotoPickerComponent.propDecorators = {
        "framework": [{ type: Input },],
        "url": [{ type: Input },],
        "path": [{ type: Input },],
        "storageOptions": [{ type: Input },],
        "picked": [{ type: Output },],
        "loaded": [{ type: Output },],
        "uploaded": [{ type: Output },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
PhotoPickerComponent.ɵfac = function PhotoPickerComponent_Factory(t) { return new (t || PhotoPickerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
PhotoPickerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PhotoPickerComponent, selectors: [["amplify-photo-picker"]], viewQuery: function PhotoPickerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DynamicComponentDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.componentHost = _t.first);
    } }, inputs: { framework: "framework", url: "url", path: "path", storageOptions: "storageOptions" }, outputs: { picked: "picked", loaded: "loaded", uploaded: "uploaded" }, decls: 2, vars: 0, consts: [[1, "amplify-component"], ["component-host", ""]], template: function PhotoPickerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, PhotoPickerComponent_ng_template_1_Template, 0, 0, "ng-template", 1);
        ɵngcc0.ɵɵelementEnd();
    } }, directives: [ɵngcc1.DynamicComponentDirective], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PhotoPickerComponent, [{
        type: Component,
        args: [{
                selector: 'amplify-photo-picker',
                template: "\n\t\t<div class=\"amplify-component\">\n\t\t\t<ng-template component-host></ng-template>\n\t\t</div>\n\t"
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }]; }, { picked: [{
            type: Output
        }], loaded: [{
            type: Output
        }], uploaded: [{
            type: Output
        }], framework: [{
            type: Input
        }], url: [{
            type: Input
        }], path: [{
            type: Input
        }], storageOptions: [{
            type: Input
        }], componentHost: [{
            type: ViewChild,
            args: [DynamicComponentDirective]
        }] }); })();
    return PhotoPickerComponent;
}());
export { PhotoPickerComponent };
