// aws-amplify-angular/__ivy_ngcc__/dist/src/components/interactions/chatbot/chatbot.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function ChatbotComponentIonic_ion_row_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-row");
    ɵngcc0.ɵɵelementStart(1, "ion-col");
    ɵngcc0.ɵɵelementStart(2, "ion-row");
    ɵngcc0.ɵɵelementStart(3, "ion-col", 9);
    ɵngcc0.ɵɵelementStart(4, "div");
    ɵngcc0.ɵɵelementStart(5, "ion-chip", 10);
    ɵngcc0.ɵɵelementStart(6, "ion-label");
    ɵngcc0.ɵɵtext(7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(8, "ion-col", 11);
    ɵngcc0.ɵɵtext(9, " \u00A0 ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(7);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.chatTitle);
} }
function ChatbotComponentIonic_ion_row_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-row");
    ɵngcc0.ɵɵelementStart(1, "ion-col");
    ɵngcc0.ɵɵelementStart(2, "ion-row");
    ɵngcc0.ɵɵelementStart(3, "ion-col", 9);
    ɵngcc0.ɵɵelementStart(4, "div");
    ɵngcc0.ɵɵtext(5, " \u00A0 ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "ion-col", 11);
    ɵngcc0.ɵɵelementStart(7, "ion-chip", 12);
    ɵngcc0.ɵɵelementStart(8, "ion-label");
    ɵngcc0.ɵɵtext(9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "ion-row");
    ɵngcc0.ɵɵelementStart(11, "ion-col", 9);
    ɵngcc0.ɵɵelementStart(12, "ion-chip", 10);
    ɵngcc0.ɵɵelementStart(13, "ion-label");
    ɵngcc0.ɵɵtext(14);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(15, "ion-col", 11);
    ɵngcc0.ɵɵelementStart(16, "div");
    ɵngcc0.ɵɵtext(17, " \u00A0 ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var message_r6 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(9);
    ɵngcc0.ɵɵtextInterpolate(message_r6.me);
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(message_r6.bot);
} }
function ChatbotComponentIonic_ion_input_6_Template(rf, ctx) { if (rf & 1) {
    var _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-input", 13, 14);
    ɵngcc0.ɵɵlistener("keyup.enter", function ChatbotComponentIonic_ion_input_6_Template_ion_input_keyup_enter_0_listener() { ɵngcc0.ɵɵrestoreView(_r9); var _r7 = ɵngcc0.ɵɵreference(1); var ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.onSubmit(_r7.value); })("ionChange", function ChatbotComponentIonic_ion_input_6_Template_ion_input_ionChange_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r9); var ctx_r10 = ɵngcc0.ɵɵnextContext(); return ctx_r10.onInputChange($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx_r2.currentVoiceState);
    ɵngcc0.ɵɵproperty("value", ctx_r2.inputText)("disabled", ctx_r2.inputDisabled);
} }
function ChatbotComponentIonic_ion_input_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "ion-input", 15, 14);
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx_r3.currentVoiceState);
    ɵngcc0.ɵɵproperty("disabled", !ctx_r3.textEnabled);
} }
function ChatbotComponentIonic_ion_button_8_Template(rf, ctx) { if (rf & 1) {
    var _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 16);
    ɵngcc0.ɵɵlistener("click", function ChatbotComponentIonic_ion_button_8_Template_ion_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r13); var ctx_r12 = ɵngcc0.ɵɵnextContext(); return ctx_r12.micButtonHandler(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("disabled", ctx_r4.micButtonDisabled);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r4.micText, " ");
} }
function ChatbotComponentIonic_ion_button_9_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 17);
    ɵngcc0.ɵɵlistener("click", function ChatbotComponentIonic_ion_button_9_Template_ion_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(); return ctx_r14.inputDisabled === false || ctx_r14.onSubmit(ctx_r14.inputValue.value); });
    ɵngcc0.ɵɵtext(1, " Send ");
    ɵngcc0.ɵɵelementEnd();
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, ChangeDetectorRef, Inject } from '@angular/core';
import { ChatbotComponentCore } from './chatbot.component.core';
import { AmplifyService } from '../../../providers/amplify.service';
var template = "\n<div class=\"amplify-interactions-container\">\n\t<div class=\"amplify-form-container\">\n\t\t<ion-grid>\n\t\t\t<ion-row *ngIf=\"chatTitle\"> \n\t\t\t\t<ion-col>\n\t\t\t\t\t<ion-row>\n\t\t\t\t\t\t<ion-col align-self-start>\n\t\t\t\t\t\t\t<div>\n\t\t\t\t\t\t\t\t<ion-chip color=\"primary\">\n\t\t\t\t\t\t\t\t\t<ion-label>{{chatTitle}}</ion-label>\n\t\t\t\t\t\t\t\t</ion-chip>\n\t\t\t\t\t\t\t</div>\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t\t<ion-col align-self-end>\n\t\t\t\t\t\t\t&nbsp;\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t</ion-row>\n\t\t\t\t</ion-col>\n\t\t\t</ion-row>\n\t\t\t<ion-row *ngFor=\"let message of messages\">\n\t\t\t\t<ion-col>\n\t\t\t\t\t<ion-row>\n\t\t\t\t\t\t<ion-col align-self-start>\n\t\t\t\t\t\t\t<div>\n\t\t\t\t\t\t\t\t&nbsp;\n\t\t\t\t\t\t\t</div>\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t\t<ion-col align-self-end>\n\t\t\t\t\t\t\t<ion-chip style=\"float:right\">\n\t\t\t\t\t\t\t\t<ion-label>{{message.me}}</ion-label>\n\t\t\t\t\t\t\t</ion-chip>\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t</ion-row>\n\t\t\t\t\t<ion-row>\n\t\t\t\t\t\t<ion-col align-self-start>\n\t\t\t\t\t\t\t<ion-chip color=\"primary\">\n\t\t\t\t\t\t\t\t<ion-label>{{message.bot}}</ion-label>\n\t\t\t\t\t\t\t</ion-chip>\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t\t<ion-col align-self-end>\n\t\t\t\t\t\t\t<div>\n\t\t\t\t\t\t\t\t&nbsp;\n\t\t\t\t\t\t\t</div>\n\t\t\t\t\t\t</ion-col>\n\t\t\t\t\t</ion-row>\n\t\t\t\t</ion-col>\n\t\t\t</ion-row>\n\t\t</ion-grid>\n\t\t<div class=\"amplify-form-row\">\n\t\t    <ion-input #inputValue\n\t\t\t\ttype='text'\n\t\t        class=\"amplify-form-input amplify-form-input-interactions-ionic\"\n\t\t        placeholder=\"{{currentVoiceState}}\"\n\t\t        [value]=\"inputText\"\n\t\t        (keyup.enter)=\"onSubmit(inputValue.value)\"\n\t\t\t\t(ionChange)=\"onInputChange($event.target.value)\"\n\t\t\t\t[disabled]=\"inputDisabled\"\n\t\t\t\t*ngIf=\"textEnabled\"></ion-input>\n\t\t\t<ion-input #inputValue\n\t\t\t\ttype='text'\n\t\t\t\tclass=\"amplify-form-input amplify-form-input-interactions-ionic\"\n\t\t\t\tplaceholder=\"{{currentVoiceState}}\"\n\t\t\t\t[disabled]=\"!textEnabled\"\n\t\t\t\t*ngIf=\"!textEnabled\"></ion-input>\n\t\t\t<ion-button \n\t\t\t\texpand=\"block\"\n\t\t\t\t*ngIf=\"voiceEnabled\"\n\t\t\t\tng-style=\"{float: 'right'}\"\n\t\t\t\t(click)=\"micButtonHandler()\"\n\t\t\t\t[disabled]=\"micButtonDisabled\"\n\t\t\t>\n\t\t\t\t{{micText}}\n\t\t\t</ion-button>\n\t\t\t<ion-button\n\t\t\t\texpand=\"block\"\n\t\t\t\t*ngIf=\"textEnabled\"\n\t\t\t\tng-style=\"{float: 'right'}\"\n\t\t\t\t(click)=\"inputDisabled === false || onSubmit(inputValue.value)\"\n\t\t\t>\n\t\t\t\tSend\n\t\t\t</ion-button>\n\t\t</div>\n\t</div>\n</div>\n";
var ChatbotComponentIonic = /** @class */ (function (_super) {
    __extends(ChatbotComponentIonic, _super);
    function ChatbotComponentIonic(ref, amplifyService) {
        var _this = _super.call(this, ref, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    ChatbotComponentIonic.ctorParameters = function () { return [
        { type: ChangeDetectorRef, },
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
ChatbotComponentIonic.ɵfac = function ChatbotComponentIonic_Factory(t) { return new (t || ChatbotComponentIonic)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
ChatbotComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ChatbotComponentIonic, selectors: [["amplify-interactions-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 10, vars: 6, consts: [[1, "amplify-interactions-container"], [1, "amplify-form-container"], [4, "ngIf"], [4, "ngFor", "ngForOf"], [1, "amplify-form-row"], ["type", "text", "class", "amplify-form-input amplify-form-input-interactions-ionic", 3, "placeholder", "value", "disabled", "keyup.enter", "ionChange", 4, "ngIf"], ["type", "text", "class", "amplify-form-input amplify-form-input-interactions-ionic", 3, "placeholder", "disabled", 4, "ngIf"], ["expand", "block", "ng-style", "{float: 'right'}", 3, "disabled", "click", 4, "ngIf"], ["expand", "block", "ng-style", "{float: 'right'}", 3, "click", 4, "ngIf"], ["align-self-start", ""], ["color", "primary"], ["align-self-end", ""], [2, "float", "right"], ["type", "text", 1, "amplify-form-input", "amplify-form-input-interactions-ionic", 3, "placeholder", "value", "disabled", "keyup.enter", "ionChange"], ["inputValue", ""], ["type", "text", 1, "amplify-form-input", "amplify-form-input-interactions-ionic", 3, "placeholder", "disabled"], ["expand", "block", "ng-style", "{float: 'right'}", 3, "disabled", "click"], ["expand", "block", "ng-style", "{float: 'right'}", 3, "click"]], template: function ChatbotComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "ion-grid");
        ɵngcc0.ɵɵtemplate(3, ChatbotComponentIonic_ion_row_3_Template, 10, 1, "ion-row", 2);
        ɵngcc0.ɵɵtemplate(4, ChatbotComponentIonic_ion_row_4_Template, 18, 2, "ion-row", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 4);
        ɵngcc0.ɵɵtemplate(6, ChatbotComponentIonic_ion_input_6_Template, 2, 3, "ion-input", 5);
        ɵngcc0.ɵɵtemplate(7, ChatbotComponentIonic_ion_input_7_Template, 2, 2, "ion-input", 6);
        ɵngcc0.ɵɵtemplate(8, ChatbotComponentIonic_ion_button_8_Template, 2, 2, "ion-button", 7);
        ɵngcc0.ɵɵtemplate(9, ChatbotComponentIonic_ion_button_9_Template, 2, 0, "ion-button", 8);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", ctx.chatTitle);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.messages);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.textEnabled);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.textEnabled);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.voiceEnabled);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.textEnabled);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChatbotComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-interactions-ionic',
                template: template
            }]
    }], function () { return [{ type: ɵngcc0.ChangeDetectorRef }, { type: AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return ChatbotComponentIonic;
}(ChatbotComponentCore));
export { ChatbotComponentIonic };
