// aws-amplify-angular/__ivy_ngcc__/dist/src/directives/dynamic.component.directive.js
import { Directive, ViewContainerRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var DynamicComponentDirective = /** @class */ (function () {
    function DynamicComponentDirective(viewContainerRef) {
        this.viewContainerRef = viewContainerRef;
    }
    /** @nocollapse */
    DynamicComponentDirective.ctorParameters = function () { return [
        { type: ViewContainerRef, },
    ]; };
DynamicComponentDirective.ɵfac = function DynamicComponentDirective_Factory(t) { return new (t || DynamicComponentDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
DynamicComponentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DynamicComponentDirective, selectors: [["", "component-host", ""]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DynamicComponentDirective, [{
        type: Directive,
        args: [{
                selector: '[component-host]'
            }]
    }], function () { return [{ type: ɵngcc0.ViewContainerRef }]; }, null); })();
    return DynamicComponentDirective;
}());
export { DynamicComponentDirective };
