// aws-amplify-angular/__ivy_ngcc__/dist/src/components/storage/s3-image-component/s3-image.factory.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, ViewChild, ComponentFactoryResolver, Output, EventEmitter, } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { S3ImageComponentIonic } from './s3-image.component.ionic';
import { S3ImageComponentCore } from './s3-image.component.core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../directives/dynamic.component.directive';

function S3ImageComponent_ng_template_1_Template(rf, ctx) { }
var S3ImageComponent = /** @class */ (function () {
    function S3ImageComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.selected = new EventEmitter();
    }
    S3ImageComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    S3ImageComponent.prototype.ngOnDestroy = function () { };
    S3ImageComponent.prototype.loadComponent = function () {
        var _this = this;
        var imageComponent = this.framework && this.framework.toLowerCase() === 'ionic'
            ? new ComponentMount(S3ImageComponentIonic, {
                path: this.path,
                options: this.options,
            })
            : new ComponentMount(S3ImageComponentCore, {
                path: this.path,
                options: this.options,
            });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(imageComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = imageComponent.data;
        componentRef.instance.selected.subscribe(function (e) {
            _this.selected.emit(e);
        });
    };
    /** @nocollapse */
    S3ImageComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    S3ImageComponent.propDecorators = {
        "framework": [{ type: Input },],
        "path": [{ type: Input },],
        "options": [{ type: Input },],
        "selected": [{ type: Output },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
S3ImageComponent.ɵfac = function S3ImageComponent_Factory(t) { return new (t || S3ImageComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
S3ImageComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: S3ImageComponent, selectors: [["amplify-s3-image"]], viewQuery: function S3ImageComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DynamicComponentDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.componentHost = _t.first);
    } }, inputs: { framework: "framework", path: "path", options: "options" }, outputs: { selected: "selected" }, decls: 2, vars: 0, consts: [["component-host", ""]], template: function S3ImageComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵtemplate(1, S3ImageComponent_ng_template_1_Template, 0, 0, "ng-template", 0);
        ɵngcc0.ɵɵelementEnd();
    } }, directives: [ɵngcc1.DynamicComponentDirective], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(S3ImageComponent, [{
        type: Component,
        args: [{
                selector: 'amplify-s3-image',
                template: "\n\t\t<div>\n\t\t\t<ng-template component-host></ng-template>\n\t\t</div>\n\t"
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }]; }, { selected: [{
            type: Output
        }], framework: [{
            type: Input
        }], path: [{
            type: Input
        }], options: [{
            type: Input
        }], componentHost: [{
            type: ViewChild,
            args: [DynamicComponentDirective]
        }] }); })();
    return S3ImageComponent;
}());
export { S3ImageComponent };
