// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/username-field-component/username-field.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../phone-field-component/phone-field.component.ionic';

function UsernameFieldComponentIonic_ion_item_0_Template(rf, ctx) { if (rf & 1) {
    var _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 1);
    ɵngcc0.ɵɵelementStart(1, "ion-label", 2);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "ion-input", 3, 4);
    ɵngcc0.ɵɵlistener("input", function UsernameFieldComponentIonic_ion_item_0_Template_ion_input_input_3_listener($event) { ɵngcc0.ɵɵrestoreView(_r5); var ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.setEmail($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.amplifyService.i18n().get("Email *"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx_r0.amplifyService.i18n().get(ctx_r0.getPlaceholder()));
} }
function UsernameFieldComponentIonic_ion_item_1_Template(rf, ctx) { if (rf & 1) {
    var _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 1);
    ɵngcc0.ɵɵelementStart(1, "amplify-auth-phone-field-ionic", 5);
    ɵngcc0.ɵɵlistener("phoneFieldChanged", function UsernameFieldComponentIonic_ion_item_1_Template_amplify_auth_phone_field_ionic_phoneFieldChanged_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); var ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.onPhoneFieldChanged($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} }
function UsernameFieldComponentIonic_ion_item_2_Template(rf, ctx) { if (rf & 1) {
    var _r10 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 1);
    ɵngcc0.ɵɵelementStart(1, "ion-label", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "ion-input", 7, 8);
    ɵngcc0.ɵɵlistener("input", function UsernameFieldComponentIonic_ion_item_2_Template_ion_input_input_3_listener($event) { ɵngcc0.ɵɵrestoreView(_r10); var ctx_r9 = ɵngcc0.ɵɵnextContext(); return ctx_r9.setUsername($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1("", ctx_r2.amplifyService.i18n().get(ctx_r2.getUsernameLabel()), " *");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("placeholder", ctx_r2.amplifyService.i18n().get(ctx_r2.getPlaceholder()));
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { UsernameFieldComponentCore } from './username-field.component.core';
import { AmplifyService } from '../../../providers/amplify.service';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n\n    <ion-item lines=\"none\" *ngIf=\"this._usernameAttributes === 'email'\">\n        <ion-label class=\"amplify-input-label amplify-input-label-ionic\" for=\"emailField\" position=\"stacked\">{{ this.amplifyService.i18n().get('Email *') }}</ion-label>\n        <ion-input type=\"text\"\n            #emailField\n            class=\"amplify-form-input\"\n            type=\"email\"\n            placeholder=\"{{ this.amplifyService.i18n().get(this.getPlaceholder()) }}\"\n            (input)=\"setEmail($event.target.value)\"\n            data-test=\"" + auth.genericAttrs.emailInput + "\"\n        ></ion-input>\n    </ion-item> \n    <ion-item lines=\"none\" *ngIf=\"this._usernameAttributes === 'phone_number'\">\n       <amplify-auth-phone-field-ionic\n            (phoneFieldChanged)=\"onPhoneFieldChanged($event)\"\n        ></amplify-auth-phone-field-ionic>\n    </ion-item>\n    <ion-item lines=\"none\" *ngIf=\"this._usernameAttributes !== 'email' && this._usernameAttributes !== 'phone_number'\">\n        <ion-label class=\"amplify-input-label amplify-input-label-ionic\" for=\"usernameField\" position=\"stacked\">{{ this.amplifyService.i18n().get(getUsernameLabel()) }} *</ion-label>\n        <ion-input type=\"text\"\n            #usernameField\n            class=\"amplify-form-input\"\n            type=\"text\"\n            placeholder=\"{{ this.amplifyService.i18n().get(this.getPlaceholder()) }}\"\n            (input)=\"setUsername($event.target.value)\"\n            data-test=\"" + auth.genericAttrs.usernameInput + "\"\n        ></ion-input>\n    </ion-item>\n\n";
var UsernameFieldComponentIonic = /** @class */ (function (_super) {
    __extends(UsernameFieldComponentIonic, _super);
    function UsernameFieldComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    UsernameFieldComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
UsernameFieldComponentIonic.ɵfac = function UsernameFieldComponentIonic_Factory(t) { return new (t || UsernameFieldComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
UsernameFieldComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: UsernameFieldComponentIonic, selectors: [["amplify-auth-username-field-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 3, consts: [["lines", "none", 4, "ngIf"], ["lines", "none"], ["for", "emailField", "position", "stacked", 1, "amplify-input-label", "amplify-input-label-ionic"], ["type", "text", "type", "email", "data-test", "email-input", 1, "amplify-form-input", 3, "placeholder", "input"], ["emailField", ""], [3, "phoneFieldChanged"], ["for", "usernameField", "position", "stacked", 1, "amplify-input-label", "amplify-input-label-ionic"], ["type", "text", "type", "text", "data-test", "username-input", 1, "amplify-form-input", 3, "placeholder", "input"], ["usernameField", ""]], template: function UsernameFieldComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, UsernameFieldComponentIonic_ion_item_0_Template, 5, 2, "ion-item", 0);
        ɵngcc0.ɵɵtemplate(1, UsernameFieldComponentIonic_ion_item_1_Template, 2, 0, "ion-item", 0);
        ɵngcc0.ɵɵtemplate(2, UsernameFieldComponentIonic_ion_item_2_Template, 5, 2, "ion-item", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx._usernameAttributes === "email");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx._usernameAttributes === "phone_number");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx._usernameAttributes !== "email" && ctx._usernameAttributes !== "phone_number");
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.PhoneFieldComponentIonic], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(UsernameFieldComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-username-field-ionic',
                template: template
            }]
    }], function () { return [{ type: AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return UsernameFieldComponentIonic;
}(UsernameFieldComponentCore));
export { UsernameFieldComponentIonic };
