// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/sign-up-component/sign-up.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../phone-field-component/phone-field.component.ionic';

function SignUpComponentIonic_div_0_ion_item_5_ion_label_1_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1, "*");
    ɵngcc0.ɵɵelementEnd();
} }
function SignUpComponentIonic_div_0_ion_item_5_ion_label_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-label", 16);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵtemplate(2, SignUpComponentIonic_div_0_ion_item_5_ion_label_1_span_2_Template, 2, 0, "span", 17);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var field_r4 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r5 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r5.amplifyService.i18n().get(field_r4.label), " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r4.required);
} }
var _c0 = function (a0) { return { "amplify-input-invalid ": a0 }; };
function SignUpComponentIonic_div_0_ion_item_5_ion_input_2_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-input", 18, 19);
    ɵngcc0.ɵɵlistener("keyup", function SignUpComponentIonic_div_0_ion_item_5_ion_input_2_Template_ion_input_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r11 = ɵngcc0.ɵɵnextContext(3); return ctx_r11.setProp($event.target); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var field_r4 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵpropertyInterpolate("type", field_r4.type);
    ɵngcc0.ɵɵpropertyInterpolate("name", field_r4.key);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c0, field_r4.invalid))("placeholder", ctx_r6.amplifyService.i18n().get(field_r4.label));
} }
function SignUpComponentIonic_div_0_ion_item_5_ion_content_3_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-content", 20);
    ɵngcc0.ɵɵelementStart(1, "amplify-auth-phone-field-ionic", 21);
    ɵngcc0.ɵɵlistener("phoneFieldChanged", function SignUpComponentIonic_div_0_ion_item_5_ion_content_3_Template_amplify_auth_phone_field_ionic_phoneFieldChanged_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(3); return ctx_r14.onPhoneFieldChanged($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var field_r4 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r7 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("label", field_r4.label)("required", field_r4.required)("placeholder", field_r4.placeholder)("defaultCountryCode", ctx_r7.country_code);
} }
function SignUpComponentIonic_div_0_ion_item_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ion-item", 4);
    ɵngcc0.ɵɵtemplate(1, SignUpComponentIonic_div_0_ion_item_5_ion_label_1_Template, 3, 2, "ion-label", 13);
    ɵngcc0.ɵɵtemplate(2, SignUpComponentIonic_div_0_ion_item_5_ion_input_2_Template, 2, 6, "ion-input", 14);
    ɵngcc0.ɵɵtemplate(3, SignUpComponentIonic_div_0_ion_item_5_ion_content_3_Template, 2, 4, "ion-content", 15);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var field_r4 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r4.key !== "phone_number");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r4.key !== "phone_number");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", field_r4.key === "phone_number");
} }
function SignUpComponentIonic_div_0_div_15_Template(rf, ctx) { if (rf & 1) {
    var _r18 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementStart(2, "a", 22);
    ɵngcc0.ɵɵlistener("click", function SignUpComponentIonic_div_0_div_15_Template_a_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r18); var ctx_r17 = ɵngcc0.ɵɵnextContext(2); return ctx_r17.onConfirmSignUp(); });
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.amplifyService.i18n().get("Have a code?"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.amplifyService.i18n().get("Confirm"), " ");
} }
function SignUpComponentIonic_div_0_div_16_Template(rf, ctx) { if (rf & 1) {
    var _r20 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 23);
    ɵngcc0.ɵɵelementStart(1, "div", 24);
    ɵngcc0.ɵɵelementStart(2, "span", 25);
    ɵngcc0.ɵɵtext(3, "\u26A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "div", 26);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "a", 27);
    ɵngcc0.ɵɵlistener("click", function SignUpComponentIonic_div_0_div_16_Template_a_click_6_listener() { ɵngcc0.ɵɵrestoreView(_r20); var ctx_r19 = ɵngcc0.ɵɵnextContext(2); return ctx_r19.onAlertClose(); });
    ɵngcc0.ɵɵtext(7, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(ctx_r3.amplifyService.i18n().get(ctx_r3.errorMessage));
} }
function SignUpComponentIonic_div_0_Template(rf, ctx) { if (rf & 1) {
    var _r22 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵelementStart(2, "div", 3);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "ion-list", 4);
    ɵngcc0.ɵɵtemplate(5, SignUpComponentIonic_div_0_ion_item_5_Template, 4, 3, "ion-item", 5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "div", 6);
    ɵngcc0.ɵɵelementStart(7, "div", 7);
    ɵngcc0.ɵɵelementStart(8, "ion-button", 8);
    ɵngcc0.ɵɵlistener("click", function SignUpComponentIonic_div_0_Template_ion_button_click_8_listener() { ɵngcc0.ɵɵrestoreView(_r22); var ctx_r21 = ɵngcc0.ɵɵnextContext(); return ctx_r21.onSignUp(); });
    ɵngcc0.ɵɵtext(9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(10, "div", 7);
    ɵngcc0.ɵɵelementStart(11, "div", 9);
    ɵngcc0.ɵɵtext(12);
    ɵngcc0.ɵɵelementStart(13, "a", 10);
    ɵngcc0.ɵɵlistener("click", function SignUpComponentIonic_div_0_Template_a_click_13_listener() { ɵngcc0.ɵɵrestoreView(_r22); var ctx_r23 = ɵngcc0.ɵɵnextContext(); return ctx_r23.onSignIn(); });
    ɵngcc0.ɵɵtext(14);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(15, SignUpComponentIonic_div_0_div_15_Template, 4, 2, "div", 11);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(16, SignUpComponentIonic_div_0_div_16_Template, 8, 1, "div", 12);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get(ctx_r0.header), "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r0.signUpFields);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.amplifyService.i18n().get("Create Account"));
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Have an account?"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Sign In"), " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.shouldHide("SignUp"));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.errorMessage);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { SignUpComponentCore } from './sign-up.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div\n  class=\"amplify-authenticator\"\n  *ngIf=\"_show\"\n  data-test=\"" + auth.signUp.section + "\"\n  >\n  <div class=\"amplify-form-body\" data-test=\"" + auth.signUp.bodySection + "\">\n    <div class=\"amplify-form-header\" data-test=\"" + auth.signUp.headerSection + "\">\n      {{ this.amplifyService.i18n().get(this.header) }}</div>\n    <ion-list lines=\"none\">\n      <ion-item lines=\"none\" *ngFor=\"let field of signUpFields\">\n        <ion-label class=\"amplify-input-label\"\n        position=\"stacked\"\n        *ngIf=\"field.key !== 'phone_number'\"\n        >\n          {{ this.amplifyService.i18n().get(field.label) }}\n          <span *ngIf=\"field.required\">*</span>\n        </ion-label>\n        <ion-input\n          [ngClass]=\"{'amplify-input-invalid ': field.invalid}\"\n          *ngIf=\"field.key !== 'phone_number'\"\n          #{{field.key}}\n          type=\"text\"\n          class=\"amplify-form-input\"\n          type={{field.type}}\n          [placeholder]=\"this.amplifyService.i18n().get(field.label)\"\n          (keyup)=\"setProp($event.target)\"\n          name={{field.key}}\n          data-test=\"" + auth.signUp.nonPhoneNumberInput + "\"\n        ></ion-input>\n        <ion-content *ngIf=\"field.key === 'phone_number'\" class=\"amplify-phone-ion-content\">\n          <amplify-auth-phone-field-ionic\n            [label]=\"field.label\"\n            [required]=\"field.required\"\n            [placeholder]=\"field.placeholder\"\n            [defaultCountryCode]=\"country_code\"\n            (phoneFieldChanged)=\"onPhoneFieldChanged($event)\"\n          ></amplify-auth-phone-field-ionic>\n        </ion-content>\n      </ion-item>\n    </ion-list>\n    <div class=\"amplify-form-actions\">\n      <div class=\"amplify-form-row\">\n        <ion-button expand=\"block\" color=\"primary\"\n          (click)=\"onSignUp()\"\n          data-test=\"" + auth.signUp.createAccountButton + "\"\n        >{{ this.amplifyService.i18n().get('Create Account') }}</ion-button>\n      </div>\n      <div class=\"amplify-form-row\">\n        <div class=\"amplify-form-signup\">\n          {{ this.amplifyService.i18n().get('Have an account?') }}\n          <a class=\"amplify-form-link\" (click)=\"onSignIn()\" data-test=\"" + auth.signUp.signInButton + "\">\n            {{ this.amplifyService.i18n().get('Sign In') }}\n          </a>\n        </div>\n        <div class=\"amplify-form-signup\" *ngIf=\"!shouldHide('SignUp')\">\n          {{ this.amplifyService.i18n().get('Have a code?') }}\n          <a\n            class=\"amplify-form-link\"\n            (click)=\"onConfirmSignUp()\"\n            data-test=\"" + auth.signUp.confirmButton + "\"\n            >\n            {{ this.amplifyService.i18n().get('Confirm') }}\n          </a>\n        </div>\n      </div>\n    </div>\n  </div>\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ this.amplifyService.i18n().get(errorMessage) }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n</div>\n";
var SignUpComponentIonic = /** @class */ (function (_super) {
    __extends(SignUpComponentIonic, _super);
    function SignUpComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    SignUpComponentIonic.prototype.setProp = function (target) {
        return (this.user[target.name] = target.value);
    };
    /** @nocollapse */
    SignUpComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
SignUpComponentIonic.ɵfac = function SignUpComponentIonic_Factory(t) { return new (t || SignUpComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
SignUpComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SignUpComponentIonic, selectors: [["amplify-auth-sign-up-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["class", "amplify-authenticator", "data-test", "sign-up-section", 4, "ngIf"], ["data-test", "sign-up-section", 1, "amplify-authenticator"], ["data-test", "sign-up-body-section", 1, "amplify-form-body"], ["data-test", "sign-up-header-section", 1, "amplify-form-header"], ["lines", "none"], ["lines", "none", 4, "ngFor", "ngForOf"], [1, "amplify-form-actions"], [1, "amplify-form-row"], ["expand", "block", "color", "primary", "data-test", "sign-up-create-account-button", 3, "click"], [1, "amplify-form-signup"], ["data-test", "sign-up-sign-in-button", 1, "amplify-form-link", 3, "click"], ["class", "amplify-form-signup", 4, "ngIf"], ["class", "amplify-alert", 4, "ngIf"], ["class", "amplify-input-label", "position", "stacked", 4, "ngIf"], ["type", "text", "class", "amplify-form-input", "data-test", "sign-up-non-phone-number-input", 3, "ngClass", "type", "placeholder", "name", "keyup", 4, "ngIf"], ["class", "amplify-phone-ion-content", 4, "ngIf"], ["position", "stacked", 1, "amplify-input-label"], [4, "ngIf"], ["type", "text", "data-test", "sign-up-non-phone-number-input", 1, "amplify-form-input", 3, "ngClass", "type", "placeholder", "name", "keyup"], ["{{field.key}}", ""], [1, "amplify-phone-ion-content"], [3, "label", "required", "placeholder", "defaultCountryCode", "phoneFieldChanged"], ["data-test", "sign-up-confirm-button", 1, "amplify-form-link", 3, "click"], [1, "amplify-alert"], [1, "amplify-alert-body"], [1, "amplify-alert-icon"], [1, "amplify-alert-message"], [1, "amplify-alert-close", 3, "click"]], template: function SignUpComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SignUpComponentIonic_div_0_Template, 17, 7, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx._show);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc1.NgClass, ɵngcc2.PhoneFieldComponentIonic], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SignUpComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-sign-up-ionic',
                template: template
            }]
    }], function () { return [{ type: AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return SignUpComponentIonic;
}(SignUpComponentCore));
export { SignUpComponentIonic };
