// aws-amplify-angular/__ivy_ngcc__/dist/src/components/storage/s3-album-component/s3-album.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function S3AlbumComponentIonic_amplify_s3_image_core_2_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "amplify-s3-image-core", 3);
    ɵngcc0.ɵɵlistener("selected", function S3AlbumComponentIonic_amplify_s3_image_core_2_Template_amplify_s3_image_core_selected_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r3); var ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.onImageSelected($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r1 = ctx.$implicit;
    ɵngcc0.ɵɵpropertyInterpolate("path", item_r1.path);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { S3AlbumComponentCore } from './s3-album.component.core';
var template = "\n<div class=\"amplify-album\">\n  <div class=\"amplify-album-container\">\n    <amplify-s3-image-core\n      class=\"amplify-image-container\"\n      *ngFor=\"let item of list\"\n      path=\"{{item.path}}\"\n      (selected)=\"onImageSelected($event)\"\n    ></amplify-s3-image-core>\n  </div>\n</div>\n";
var S3AlbumComponentIonic = /** @class */ (function (_super) {
    __extends(S3AlbumComponentIonic, _super);
    function S3AlbumComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    S3AlbumComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
S3AlbumComponentIonic.ɵfac = function S3AlbumComponentIonic_Factory(t) { return new (t || S3AlbumComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
S3AlbumComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: S3AlbumComponentIonic, selectors: [["amplify-s3-album-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [[1, "amplify-album"], [1, "amplify-album-container"], ["class", "amplify-image-container", 3, "path", "selected", 4, "ngFor", "ngForOf"], [1, "amplify-image-container", 3, "path", "selected"]], template: function S3AlbumComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtemplate(2, S3AlbumComponentIonic_amplify_s3_image_core_2_Template, 1, 1, "amplify-s3-image-core", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.list);
    } }, directives: [ɵngcc1.NgForOf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(S3AlbumComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-s3-album-ionic',
                template: template
            }]
    }], function () { return [{ type: AmplifyService }]; }, null); })();
    return S3AlbumComponentIonic;
}(S3AlbumComponentCore));
export { S3AlbumComponentIonic };
