// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/greeting-component/greeting.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function GreetingComponentIonic_div_0_ion_button_4_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 5);
    ɵngcc0.ɵɵlistener("click", function GreetingComponentIonic_div_0_ion_button_4_Template_ion_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); var ctx_r2 = ɵngcc0.ɵɵnextContext(2); return ctx_r2.onSignOut(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.amplifyService.i18n().get("Sign Out"));
} }
function GreetingComponentIonic_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "div", 3);
    ɵngcc0.ɵɵtemplate(4, GreetingComponentIonic_div_0_ion_button_4_Template, 2, 1, "ion-button", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.greeting);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.signedIn);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { GreetingComponentCore } from './greeting.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div class=\"amplify-greeting\" *ngIf=\"signedIn\">\n    <div class=\"amplify-greeting-text\">{{ greeting }}</div>\n    <div class=\"amplify-greeting-flex-spacer\"></div>\n    <ion-button\n        class=\"amplify-greeting-sign-out\"\n        size=\"small\"\n        *ngIf=\"signedIn\"\n        (click)=\"onSignOut()\"\n        data-test=\"" + auth.greeting.signOutButton + "\"\n      >{{ this.amplifyService.i18n().get('Sign Out') }}</ion-button>\n</div>\n";
var GreetingComponentIonic = /** @class */ (function (_super) {
    __extends(GreetingComponentIonic, _super);
    function GreetingComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    /** @nocollapse */
    GreetingComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
GreetingComponentIonic.ɵfac = function GreetingComponentIonic_Factory(t) { return new (t || GreetingComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
GreetingComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: GreetingComponentIonic, selectors: [["amplify-auth-greetings-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["class", "amplify-greeting", 4, "ngIf"], [1, "amplify-greeting"], [1, "amplify-greeting-text"], [1, "amplify-greeting-flex-spacer"], ["class", "amplify-greeting-sign-out", "size", "small", "data-test", "sign-out-button", 3, "click", 4, "ngIf"], ["size", "small", "data-test", "sign-out-button", 1, "amplify-greeting-sign-out", 3, "click"]], template: function GreetingComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, GreetingComponentIonic_div_0_Template, 5, 2, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.signedIn);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(GreetingComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-greetings-ionic',
                template: template
            }]
    }], function () { return [{ type: AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return GreetingComponentIonic;
}(GreetingComponentCore));
export { GreetingComponentIonic };
