// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/forgot-password-component/forgot-password.component.ionic.js
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../username-field-component/username-field.component.ionic';

function ForgotPasswordComponentIonic_div_0_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.amplifyService.i18n().get("You will receive a verification code to reset your password"), " ");
} }
function ForgotPasswordComponentIonic_div_0_ion_item_7_Template(rf, ctx) { if (rf & 1) {
    var _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 16);
    ɵngcc0.ɵɵelementStart(1, "ion-label", 17);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "ion-input", 18, 19);
    ɵngcc0.ɵɵlistener("input", function ForgotPasswordComponentIonic_div_0_ion_item_7_Template_ion_input_input_3_listener() { ɵngcc0.ɵɵrestoreView(_r9); var _r7 = ɵngcc0.ɵɵreference(4); var ctx_r8 = ɵngcc0.ɵɵnextContext(2); return ctx_r8.setCode(_r7.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r2.amplifyService.i18n().get("Code *"), " ");
} }
function ForgotPasswordComponentIonic_div_0_ion_item_8_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-item", 16);
    ɵngcc0.ɵɵelementStart(1, "ion-label", 17);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "ion-input", 20, 21);
    ɵngcc0.ɵɵlistener("input", function ForgotPasswordComponentIonic_div_0_ion_item_8_Template_ion_input_input_3_listener() { ɵngcc0.ɵɵrestoreView(_r12); var _r10 = ɵngcc0.ɵɵreference(4); var ctx_r11 = ɵngcc0.ɵɵnextContext(2); return ctx_r11.setPassword(_r10.value); })("keyup.enter", function ForgotPasswordComponentIonic_div_0_ion_item_8_Template_ion_input_keyup_enter_3_listener() { ɵngcc0.ɵɵrestoreView(_r12); var ctx_r13 = ɵngcc0.ɵɵnextContext(2); return ctx_r13.onSubmit(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r3.amplifyService.i18n().get("Password *"), " ");
} }
function ForgotPasswordComponentIonic_div_0_ion_button_11_Template(rf, ctx) { if (rf & 1) {
    var _r15 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 22);
    ɵngcc0.ɵɵlistener("click", function ForgotPasswordComponentIonic_div_0_ion_button_11_Template_ion_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r15); var ctx_r14 = ɵngcc0.ɵɵnextContext(2); return ctx_r14.onSend(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r4.amplifyService.i18n().get("Submit"), "");
} }
function ForgotPasswordComponentIonic_div_0_ion_button_12_Template(rf, ctx) { if (rf & 1) {
    var _r17 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ion-button", 23);
    ɵngcc0.ɵɵlistener("click", function ForgotPasswordComponentIonic_div_0_ion_button_12_Template_ion_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r17); var ctx_r16 = ɵngcc0.ɵɵnextContext(2); return ctx_r16.onSubmit(); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r5.amplifyService.i18n().get("Verify"));
} }
function ForgotPasswordComponentIonic_div_0_div_22_Template(rf, ctx) { if (rf & 1) {
    var _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 24);
    ɵngcc0.ɵɵelementStart(1, "div", 25);
    ɵngcc0.ɵɵelementStart(2, "span", 26);
    ɵngcc0.ɵɵtext(3, "\u26A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "div", 27);
    ɵngcc0.ɵɵtext(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "a", 28);
    ɵngcc0.ɵɵlistener("click", function ForgotPasswordComponentIonic_div_0_div_22_Template_a_click_6_listener() { ɵngcc0.ɵɵrestoreView(_r19); var ctx_r18 = ɵngcc0.ɵɵnextContext(2); return ctx_r18.onAlertClose(); });
    ɵngcc0.ɵɵtext(7, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(5);
    ɵngcc0.ɵɵtextInterpolate(ctx_r6.amplifyService.i18n().get(ctx_r6.errorMessage));
} }
function ForgotPasswordComponentIonic_div_0_Template(rf, ctx) { if (rf & 1) {
    var _r21 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵelementStart(1, "div", 2);
    ɵngcc0.ɵɵelementStart(2, "div", 3);
    ɵngcc0.ɵɵtext(3);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(4, ForgotPasswordComponentIonic_div_0_div_4_Template, 2, 1, "div", 4);
    ɵngcc0.ɵɵelementStart(5, "ion-list");
    ɵngcc0.ɵɵelementStart(6, "amplify-auth-username-field-ionic", 5);
    ɵngcc0.ɵɵlistener("usernameFieldChanged", function ForgotPasswordComponentIonic_div_0_Template_amplify_auth_username_field_ionic_usernameFieldChanged_6_listener($event) { ɵngcc0.ɵɵrestoreView(_r21); var ctx_r20 = ɵngcc0.ɵɵnextContext(); return ctx_r20.onUsernameFieldChanged($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(7, ForgotPasswordComponentIonic_div_0_ion_item_7_Template, 5, 1, "ion-item", 6);
    ɵngcc0.ɵɵtemplate(8, ForgotPasswordComponentIonic_div_0_ion_item_8_Template, 5, 1, "ion-item", 6);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "div", 7);
    ɵngcc0.ɵɵelementStart(10, "div");
    ɵngcc0.ɵɵtemplate(11, ForgotPasswordComponentIonic_div_0_ion_button_11_Template, 2, 1, "ion-button", 8);
    ɵngcc0.ɵɵtemplate(12, ForgotPasswordComponentIonic_div_0_ion_button_12_Template, 2, 1, "ion-button", 9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(13, "div", 10);
    ɵngcc0.ɵɵelementStart(14, "div", 11);
    ɵngcc0.ɵɵtext(15);
    ɵngcc0.ɵɵelementStart(16, "a", 12);
    ɵngcc0.ɵɵlistener("click", function ForgotPasswordComponentIonic_div_0_Template_a_click_16_listener() { ɵngcc0.ɵɵrestoreView(_r21); var ctx_r22 = ɵngcc0.ɵɵnextContext(); return ctx_r22.onSignIn(); });
    ɵngcc0.ɵɵtext(17);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(18, "div", 11);
    ɵngcc0.ɵɵtext(19);
    ɵngcc0.ɵɵelementStart(20, "a", 13);
    ɵngcc0.ɵɵlistener("click", function ForgotPasswordComponentIonic_div_0_Template_a_click_20_listener() { ɵngcc0.ɵɵrestoreView(_r21); var ctx_r23 = ɵngcc0.ɵɵnextContext(); return ctx_r23.onSend(); });
    ɵngcc0.ɵɵtext(21);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(22, ForgotPasswordComponentIonic_div_0_div_22_Template, 8, 1, "div", 14);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Reset your password"), " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.code_sent);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("usernameAttributes", ctx_r0._usernameAttributes);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.code_sent);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.code_sent);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", !ctx_r0.code_sent);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.code_sent);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Have an account?"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Sign In"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Lost your code?"), " ");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.amplifyService.i18n().get("Resend"), " ");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.errorMessage);
} }
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Inject } from '@angular/core';
import { AmplifyService } from '../../../providers/amplify.service';
import { ForgotPasswordComponentCore } from './forgot-password.component.core';
import { auth } from '../../../assets/data-test-attributes';
var template = "\n<div\n  class=\"amplify-authenticator amplify-authenticator-ionic\"\n  *ngIf=\"_show\"\n  data-test=\"" + auth.forgotPassword.section + "\"\n  >\n  <div class=\"amplify-form-body\" data-test=\"" + auth.forgotPassword.bodySection + "\">\n  <div\n    class=\"amplify-form-header amplify-form-header-ionic\"\n    data-test=\"" + auth.forgotPassword.headerSection + "\"\n    >\n    {{ this.amplifyService.i18n().get('Reset your password') }}\n  </div>\n  <div class=\"amplify-form-text\" *ngIf=\"!code_sent\">\n    {{ this.amplifyService.i18n().get('You will receive a verification code to reset your password') }}\n  </div>\n\n  <ion-list>\n    <amplify-auth-username-field-ionic\n      [usernameAttributes]=\"_usernameAttributes\"\n      (usernameFieldChanged)=\"onUsernameFieldChanged($event)\"\n    ></amplify-auth-username-field-ionic>\n    <ion-item lines=\"none\" *ngIf=\"code_sent\">\n      <ion-label class=\"amplify-input-label amplify-input-label-ionic\" position=\"stacked\">\n        {{ this.amplifyService.i18n().get('Code *') }}\n      </ion-label>\n      <ion-input\n        #code\n        type=\"text\"\n        class=\"amplify-form-input\"\n        (input)=\"setCode(code.value)\"\n        data-test=\"" + auth.forgotPassword.codeInput + "\"\n      ></ion-input>\n    </ion-item>\n    <ion-item lines=\"none\" *ngIf=\"code_sent\">\n      <ion-label class=\"amplify-input-label amplify-input-label-ionic\" position=\"stacked\">\n        {{ this.amplifyService.i18n().get('Password *') }}\n      </ion-label>\n      <ion-input\n        #password\n        type=\"password\"\n        class=\"amplify-form-input\"\n        (input)=\"setPassword(password.value)\"\n        (keyup.enter)=\"onSubmit()\"\n        data-test=\"" + auth.forgotPassword.newPasswordInput + "\"\n      ></ion-input>\n    </ion-item>\n  </ion-list>\n  <div class=\"amplify-form-actions\">\n    <div>\n      <ion-button expand=\"block\" color=\"primary\"\n        (click)=\"onSend()\"\n        *ngIf=\"!code_sent\"\n        data-test=\"" + auth.forgotPassword.submitButton + "\"\n      >\n        {{ this.amplifyService.i18n().get('Submit') }}</ion-button>\n      <ion-button expand=\"block\" color=\"primary\"\n      *ngIf=\"code_sent\"\n      (click)=\"onSubmit()\"\n      >{{ this.amplifyService.i18n().get('Verify') }}</ion-button>\n    </div>\n    <div class=\"amplify-form-row\">\n      <div class=\"amplify-form-signup\">\n        {{ this.amplifyService.i18n().get('Have an account?') }}\n        <a class=\"amplify-form-link\" (click)=\"onSignIn()\">\n          {{ this.amplifyService.i18n().get('Sign In') }}\n        </a>\n      </div>\n      <div class=\"amplify-form-signup\">\n        {{ this.amplifyService.i18n().get('Lost your code?') }}\n        <a\n          class=\"amplify-form-link\"\n          (click)=\"onSend()\"\n          data-test=\"" + auth.forgotPassword.resendCodeLink + "\"\n          >\n          {{ this.amplifyService.i18n().get('Resend') }}\n        </a>\n      </div>\n    </div>\n  </div>\n  </div>\n  <div class=\"amplify-alert\" *ngIf=\"errorMessage\">\n    <div class=\"amplify-alert-body\">\n      <span class=\"amplify-alert-icon\">&#9888;</span>\n      <div class=\"amplify-alert-message\">{{ this.amplifyService.i18n().get(errorMessage) }}</div>\n      <a class=\"amplify-alert-close\" (click)=\"onAlertClose()\">&times;</a>\n    </div>\n  </div>\n\n</div>\n";
var ForgotPasswordComponentIonic = /** @class */ (function (_super) {
    __extends(ForgotPasswordComponentIonic, _super);
    function ForgotPasswordComponentIonic(amplifyService) {
        var _this = _super.call(this, amplifyService) || this;
        _this.amplifyService = amplifyService;
        return _this;
    }
    ForgotPasswordComponentIonic.prototype.onCodeChange = function (val) {
        this.country_code = val;
    };
    ForgotPasswordComponentIonic.prototype.onNumberChange = function (val) {
        this.local_phone_number = val;
    };
    ForgotPasswordComponentIonic.prototype.setUsername = function (username) {
        this.username = username;
    };
    ForgotPasswordComponentIonic.prototype.setEmail = function (email) {
        this.email = email;
    };
    /** @nocollapse */
    ForgotPasswordComponentIonic.ctorParameters = function () { return [
        { type: AmplifyService, decorators: [{ type: Inject, args: [AmplifyService,] },] },
    ]; };
ForgotPasswordComponentIonic.ɵfac = function ForgotPasswordComponentIonic_Factory(t) { return new (t || ForgotPasswordComponentIonic)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
ForgotPasswordComponentIonic.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ForgotPasswordComponentIonic, selectors: [["amplify-auth-forgot-password-ionic"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["class", "amplify-authenticator amplify-authenticator-ionic", "data-test", "forgot-password-section", 4, "ngIf"], ["data-test", "forgot-password-section", 1, "amplify-authenticator", "amplify-authenticator-ionic"], ["data-test", "forgot-password-body-section", 1, "amplify-form-body"], ["data-test", "forgot-password-header-section", 1, "amplify-form-header", "amplify-form-header-ionic"], ["class", "amplify-form-text", 4, "ngIf"], [3, "usernameAttributes", "usernameFieldChanged"], ["lines", "none", 4, "ngIf"], [1, "amplify-form-actions"], ["expand", "block", "color", "primary", "data-test", "forgot-password-submit-button", 3, "click", 4, "ngIf"], ["expand", "block", "color", "primary", 3, "click", 4, "ngIf"], [1, "amplify-form-row"], [1, "amplify-form-signup"], [1, "amplify-form-link", 3, "click"], ["data-test", "forgot-password-resend-code-link", 1, "amplify-form-link", 3, "click"], ["class", "amplify-alert", 4, "ngIf"], [1, "amplify-form-text"], ["lines", "none"], ["position", "stacked", 1, "amplify-input-label", "amplify-input-label-ionic"], ["type", "text", "data-test", "forgot-password-code-input", 1, "amplify-form-input", 3, "input"], ["code", ""], ["type", "password", "data-test", "forgot-password-new-password-input", 1, "amplify-form-input", 3, "input", "keyup.enter"], ["password", ""], ["expand", "block", "color", "primary", "data-test", "forgot-password-submit-button", 3, "click"], ["expand", "block", "color", "primary", 3, "click"], [1, "amplify-alert"], [1, "amplify-alert-body"], [1, "amplify-alert-icon"], [1, "amplify-alert-message"], [1, "amplify-alert-close", 3, "click"]], template: function ForgotPasswordComponentIonic_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, ForgotPasswordComponentIonic_div_0_Template, 23, 12, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx._show);
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.UsernameFieldComponentIonic], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ForgotPasswordComponentIonic, [{
        type: Component,
        args: [{
                selector: 'amplify-auth-forgot-password-ionic',
                template: template
            }]
    }], function () { return [{ type: AmplifyService, decorators: [{
                type: Inject,
                args: [AmplifyService]
            }] }]; }, null); })();
    return ForgotPasswordComponentIonic;
}(ForgotPasswordComponentCore));
export { ForgotPasswordComponentIonic };
