// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/authenticator/authenticator.component.core.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input } from '@angular/core';
import { AmplifyService } from '../../../providers';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '../sign-in-component/sign-in.component.core';
import * as ɵngcc3 from '../sign-up-component/sign-up.component.core';
import * as ɵngcc4 from '../confirm-sign-up-component/confirm-sign-up.component.core';
import * as ɵngcc5 from '../confirm-sign-in-component/confirm-sign-in-component.core';
import * as ɵngcc6 from '../forgot-password-component/forgot-password.component.core';
import * as ɵngcc7 from '../greeting-component/greeting.component.core';
import * as ɵngcc8 from '../require-new-password-component/require-new-password.component.core';

function AuthenticatorComponentCore_amplify_auth_sign_in_core_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-sign-in-core", 5);
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r0.authState)("usernameAttributes", ctx_r0._usernameAttributes)("hide", ctx_r0.hide);
} }
function AuthenticatorComponentCore_amplify_auth_sign_up_core_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-sign-up-core", 6);
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r1.authState)("signUpConfig", ctx_r1._signUpConfig)("usernameAttributes", ctx_r1._usernameAttributes)("hide", ctx_r1.hide);
} }
function AuthenticatorComponentCore_amplify_auth_confirm_sign_up_core_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-confirm-sign-up-core", 5);
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r2.authState)("usernameAttributes", ctx_r2._usernameAttributes)("hide", ctx_r2.hide);
} }
function AuthenticatorComponentCore_amplify_auth_confirm_sign_in_core_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-confirm-sign-in-core", 7);
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r3.authState)("hide", ctx_r3.hide);
} }
function AuthenticatorComponentCore_amplify_auth_forgot_password_core_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-forgot-password-core", 5);
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r4.authState)("usernameAttributes", ctx_r4._usernameAttributes)("hide", ctx_r4.hide);
} }
function AuthenticatorComponentCore_amplify_auth_greetings_core_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-greetings-core", 8);
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r5.authState)("usernameAttributes", ctx_r5._usernameAttributes);
} }
function AuthenticatorComponentCore_amplify_auth_require_new_password_core_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "amplify-auth-require-new-password-core", 7);
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("authState", ctx_r6.authState)("hide", ctx_r6.hide);
} }
var template = "\n  <div class=\"amplify-authenticator\">\n    <amplify-auth-sign-in-core\n      *ngIf=\"!shouldHide('SignIn')\"\n      [authState]=\"authState\"\n      [usernameAttributes]=\"_usernameAttributes\"\n      [hide]=\"hide\"\n    ></amplify-auth-sign-in-core>\n\n    <amplify-auth-sign-up-core\n      *ngIf=\"!shouldHide('SignUp')\"\n      [authState]=\"authState\"\n      [signUpConfig]=\"_signUpConfig\"\n      [usernameAttributes]=\"_usernameAttributes\"\n      [hide]=\"hide\"\n    ></amplify-auth-sign-up-core>\n\n    <amplify-auth-confirm-sign-up-core\n      *ngIf=\"!shouldHide('ConfirmSignUp')\"\n      [authState]=\"authState\"\n      [usernameAttributes]=\"_usernameAttributes\"\n      [hide]=\"hide\"\n    ></amplify-auth-confirm-sign-up-core>\n\n    <amplify-auth-confirm-sign-in-core\n      *ngIf=\"!shouldHide('ConfirmSignIn')\"\n      [authState]=\"authState\"\n      [hide]=\"hide\"\n    ></amplify-auth-confirm-sign-in-core>\n\n    <amplify-auth-forgot-password-core\n      *ngIf=\"!shouldHide('ForgotPassword')\"\n      [authState]=\"authState\"\n      [usernameAttributes]=\"_usernameAttributes\"\n      [hide]=\"hide\"\n    ></amplify-auth-forgot-password-core>\n\n    <amplify-auth-greetings-core\n    *ngIf=\"!shouldHide('Greetings')\"\n    [authState]=\"authState\"\n    [usernameAttributes]=\"_usernameAttributes\"\n    ></amplify-auth-greetings-core>\n\n     <amplify-auth-require-new-password-core\n      *ngIf=\"!shouldHide('RequireNewPassword')\"\n      [authState]=\"authState\"\n      [hide]=\"hide\"\n    ></amplify-auth-require-new-password-core>\n  </div>\n";
var AuthenticatorComponentCore = /** @class */ (function () {
    function AuthenticatorComponentCore(amplifyService) {
        this.amplifyService = amplifyService;
        this.authState = {
            state: 'loading',
            user: null,
        };
        this._signUpConfig = {};
        this._usernameAttributes = 'username';
        this.hide = [];
        this.subscribe();
    }
    AuthenticatorComponentCore.prototype.ngOnInit = function () {
        var _this = this;
        if (!this.amplifyService.auth()) {
            throw new Error('Auth module not registered on AmplifyService provider');
        }
        else {
            var loadStatus = this.amplifyService
                .auth()
                .currentAuthenticatedUser()
                .then(function (user) {
                if (_this.authState.state === 'loading' && user) {
                    _this.amplifyService.setAuthState({ state: 'signedIn', user: user });
                }
            })
                .catch(function (e) {
                if (_this.authState.state === 'loading') {
                    _this.amplifyService.setAuthState({ state: 'signIn', user: null });
                }
            });
        }
    };
    Object.defineProperty(AuthenticatorComponentCore.prototype, "data", {
        set: function (data) {
            if (data.signUpConfig) {
                this._signUpConfig = data.signUpConfig;
            }
            if (data.hide) {
                this.hide = data.hide;
            }
            this._usernameAttributes =
                data.usernameAttributes || this._usernameAttributes || 'username';
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AuthenticatorComponentCore.prototype, "signUpConfig", {
        set: function (signUpConfig) {
            this._signUpConfig = signUpConfig;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(AuthenticatorComponentCore.prototype, "usernameAttributes", {
        set: function (usernameAttributes) {
            this._usernameAttributes = usernameAttributes || 'username';
        },
        enumerable: true,
        configurable: true
    });
    AuthenticatorComponentCore.prototype.subscribe = function () {
        var _this = this;
        this.amplifyService.authStateChange$.subscribe(function (state) {
            _this.authState = state;
        }, function () {
            _this.authState = {
                state: 'signIn',
                user: null,
            };
        });
    };
    AuthenticatorComponentCore.prototype.shouldHide = function (comp) {
        return this.hide.filter(function (item) { return item === comp; }).length > 0;
    };
    /** @nocollapse */
    AuthenticatorComponentCore.ctorParameters = function () { return [
        { type: AmplifyService, },
    ]; };
    AuthenticatorComponentCore.propDecorators = {
        "hide": [{ type: Input },],
        "data": [{ type: Input },],
        "signUpConfig": [{ type: Input },],
        "usernameAttributes": [{ type: Input },],
    };
AuthenticatorComponentCore.ɵfac = function AuthenticatorComponentCore_Factory(t) { return new (t || AuthenticatorComponentCore)(ɵngcc0.ɵɵdirectiveInject(AmplifyService)); };
AuthenticatorComponentCore.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AuthenticatorComponentCore, selectors: [["amplify-authenticator-core"]], inputs: { hide: "hide", data: "data", signUpConfig: "signUpConfig", usernameAttributes: "usernameAttributes" }, decls: 8, vars: 7, consts: [[1, "amplify-authenticator"], [3, "authState", "usernameAttributes", "hide", 4, "ngIf"], [3, "authState", "signUpConfig", "usernameAttributes", "hide", 4, "ngIf"], [3, "authState", "hide", 4, "ngIf"], [3, "authState", "usernameAttributes", 4, "ngIf"], [3, "authState", "usernameAttributes", "hide"], [3, "authState", "signUpConfig", "usernameAttributes", "hide"], [3, "authState", "hide"], [3, "authState", "usernameAttributes"]], template: function AuthenticatorComponentCore_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, AuthenticatorComponentCore_amplify_auth_sign_in_core_1_Template, 1, 3, "amplify-auth-sign-in-core", 1);
        ɵngcc0.ɵɵtemplate(2, AuthenticatorComponentCore_amplify_auth_sign_up_core_2_Template, 1, 4, "amplify-auth-sign-up-core", 2);
        ɵngcc0.ɵɵtemplate(3, AuthenticatorComponentCore_amplify_auth_confirm_sign_up_core_3_Template, 1, 3, "amplify-auth-confirm-sign-up-core", 1);
        ɵngcc0.ɵɵtemplate(4, AuthenticatorComponentCore_amplify_auth_confirm_sign_in_core_4_Template, 1, 2, "amplify-auth-confirm-sign-in-core", 3);
        ɵngcc0.ɵɵtemplate(5, AuthenticatorComponentCore_amplify_auth_forgot_password_core_5_Template, 1, 3, "amplify-auth-forgot-password-core", 1);
        ɵngcc0.ɵɵtemplate(6, AuthenticatorComponentCore_amplify_auth_greetings_core_6_Template, 1, 2, "amplify-auth-greetings-core", 4);
        ɵngcc0.ɵɵtemplate(7, AuthenticatorComponentCore_amplify_auth_require_new_password_core_7_Template, 1, 2, "amplify-auth-require-new-password-core", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("SignIn"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("SignUp"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ConfirmSignUp"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ConfirmSignIn"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("ForgotPassword"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("Greetings"));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.shouldHide("RequireNewPassword"));
    } }, directives: [ɵngcc1.NgIf, ɵngcc2.SignInComponentCore, ɵngcc3.SignUpComponentCore, ɵngcc4.ConfirmSignUpComponentCore, ɵngcc5.ConfirmSignInComponentCore, ɵngcc6.ForgotPasswordComponentCore, ɵngcc7.GreetingComponentCore, ɵngcc8.RequireNewPasswordComponentCore], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AuthenticatorComponentCore, [{
        type: Component,
        args: [{
                selector: 'amplify-authenticator-core',
                template: template
            }]
    }], function () { return [{ type: AmplifyService }]; }, { hide: [{
            type: Input
        }], data: [{
            type: Input
        }], signUpConfig: [{
            type: Input
        }], usernameAttributes: [{
            type: Input
        }] }); })();
    return AuthenticatorComponentCore;
}());
export { AuthenticatorComponentCore };
