// aws-amplify-angular/__ivy_ngcc__/dist/src/components/authenticator/authenticator/authenticator.factory.js
// tslint:disable
/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
// tslint:enable
import { Component, Input, ViewChild, ComponentFactoryResolver, } from '@angular/core';
import { DynamicComponentDirective } from '../../../directives/dynamic.component.directive';
import { ComponentMount } from '../../component.mount';
import { AuthenticatorIonicComponent } from './authenticator.component.ionic';
import { AuthenticatorComponentCore } from './authenticator.component.core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../../directives/dynamic.component.directive';

function AuthenticatorComponent_ng_template_1_Template(rf, ctx) { }
var AuthenticatorComponent = /** @class */ (function () {
    function AuthenticatorComponent(componentFactoryResolver) {
        this.componentFactoryResolver = componentFactoryResolver;
        this.hide = [];
        this.usernameAttributes = 'username';
    }
    AuthenticatorComponent.prototype.ngOnInit = function () {
        this.loadComponent();
    };
    AuthenticatorComponent.prototype.ngOnDestroy = function () { };
    AuthenticatorComponent.prototype.loadComponent = function () {
        var authComponent = this.framework && this.framework.toLowerCase() === 'ionic'
            ? new ComponentMount(AuthenticatorIonicComponent, {
                hide: this.hide,
                signUpConfig: this.signUpConfig,
                usernameAttributes: this.usernameAttributes,
            })
            : new ComponentMount(AuthenticatorComponentCore, {
                hide: this.hide,
                signUpConfig: this.signUpConfig,
                usernameAttributes: this.usernameAttributes,
            });
        var componentFactory = this.componentFactoryResolver.resolveComponentFactory(authComponent.component);
        var viewContainerRef = this.componentHost.viewContainerRef;
        viewContainerRef.clear();
        var componentRef = viewContainerRef.createComponent(componentFactory);
        componentRef.instance.data = authComponent.data;
    };
    /** @nocollapse */
    AuthenticatorComponent.ctorParameters = function () { return [
        { type: ComponentFactoryResolver, },
    ]; };
    AuthenticatorComponent.propDecorators = {
        "framework": [{ type: Input },],
        "hide": [{ type: Input },],
        "signUpConfig": [{ type: Input },],
        "usernameAttributes": [{ type: Input },],
        "componentHost": [{ type: ViewChild, args: [DynamicComponentDirective,] },],
    };
AuthenticatorComponent.ɵfac = function AuthenticatorComponent_Factory(t) { return new (t || AuthenticatorComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
AuthenticatorComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AuthenticatorComponent, selectors: [["amplify-authenticator"]], viewQuery: function AuthenticatorComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DynamicComponentDirective, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.componentHost = _t.first);
    } }, inputs: { hide: "hide", usernameAttributes: "usernameAttributes", framework: "framework", signUpConfig: "signUpConfig" }, decls: 2, vars: 0, consts: [[1, "amplify-component"], ["component-host", ""]], template: function AuthenticatorComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, AuthenticatorComponent_ng_template_1_Template, 0, 0, "ng-template", 1);
        ɵngcc0.ɵɵelementEnd();
    } }, directives: [ɵngcc1.DynamicComponentDirective], encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AuthenticatorComponent, [{
        type: Component,
        args: [{
                selector: 'amplify-authenticator',
                template: "\n\t\t<div class=\"amplify-component\">\n\t\t\t<ng-template component-host></ng-template>\n\t\t</div>\n\t"
            }]
    }], function () { return [{ type: ɵngcc0.ComponentFactoryResolver }]; }, { hide: [{
            type: Input
        }], usernameAttributes: [{
            type: Input
        }], framework: [{
            type: Input
        }], signUpConfig: [{
            type: Input
        }], componentHost: [{
            type: ViewChild,
            args: [DynamicComponentDirective]
        }] }); })();
    return AuthenticatorComponent;
}());
export { AuthenticatorComponent };
